backend/windmill-worker/src/common.rs

```diff
@@ -732,6 +711,62 @@ pub async fn start_child_process(mut cmd: Command, executable: &str) -> Result<C
         .map_err(|err| tentatively_improve_error(Error::IoErr(err), executable));
 }
 
+async fn resolve_job_timeout(
+    db: &Pool<Postgres>,
+    w_id: &str,
+    job_id: Uuid,
+    custom_timeout_secs: Option<i32>,
+) -> (Duration, Option<String>) {
+    let mut warn_msg: Option<String> = None;
+    #[cfg(feature = "enterprise")]
+    let cloud_premium_workspace = *CLOUD_HOSTED
+        && sqlx::query_scalar!("SELECT premium FROM workspace WHERE id = $1", w_id)
+            .fetch_one(db)
+            .await
+            .map_err(|e| {
+                tracing::error!(%e, "error getting premium workspace for job {job_id}: {e}");
+            })
+            .unwrap_or(false);
+    #[cfg(not(feature = "enterprise"))]
+    let cloud_premium_workspace = false;
+
+    // compute global max timeout
+    let global_max_timeout_duration = if cloud_premium_workspace {
+        *MAX_TIMEOUT_DURATION * 6 //30mins
+    } else {
+        *MAX_TIMEOUT_DURATION
+    };
+
+    // compute default timeout
+    let default_timeout = match JOB_DEFAULT_TIMEOUT.read().await.clone() {
+        0 => global_max_timeout_duration,
+        default_timeout_secs
+            if Duration::from_secs(default_timeout_secs as u64) < global_max_timeout_duration =>
+        {
+            Duration::from_secs(default_timeout_secs as u64)
+        }
+        default_timeout_secs => {
+            warn_msg = Some(format!("WARNING: Default job timeout of {default_timeout_secs} seconds was greater than the maximum timeout. It will be ignored and the global max timeout will be used instead"));
+            tracing::warn!(warn_msg);
+            global_max_timeout_duration
+        }
+    };
```

Comment:
this is computed even if custom_timeout is set which is a waste

Reply:
nvm

Reply:
actually no, first comment is correct, you shouldn't have to read JOB_DEFAULT_TIMEOUT if custom_timeout is set